packages/playwright-core/src/server/dispatchers/pageDispatcher.ts

```diff
@@ -67,7 +77,7 @@ export class PageDispatcher extends Dispatcher<Page, channels.PageChannel> imple
     this.addObjectListener(Page.Events.Crash, () => this._dispatchEvent('crash'));
     this.addObjectListener(Page.Events.Dialog, dialog => this._dispatchEvent('dialog', { dialog: new DialogDispatcher(this._scope, dialog) }));
     this.addObjectListener(Page.Events.Download, (download: Download) => {
-      this._dispatchEvent('download', { url: download.url, suggestedFilename: download.suggestedFilename(), artifact: new ArtifactDispatcher(scope, download.artifact) });
+      this._dispatchEvent('download', { url: download.url, suggestedFilename: download.suggestedFilename(), artifact: new ArtifactDispatcher(this._scope, download.artifact) });
```

Comment:
It seems like `download.saveAs()` could be called after the page is closed? Especially for the popup pages. So perhaps `parentScope` is better.